File: languages/php/Bloock/Entity/Key/ManagedCertificateParams.php
<?php

namespace Bloock\Entity\Key;

/**
 * Represents parameters for creating a managed certificate.
 */
class ManagedCertificateParams
{
    /**
     * Is the type of the key.
     * @var string
     */
    public string $keyType;
    /**
     * Represents the subject details of the certificate.
     * @var SubjectCertificateParams
     */
    public SubjectCertificateParams $subjectParams;
    /**
     * Is the number of months until the certificate expiration.
     * @var int
     */
    public int $expiration;

    /**
     * Constructs a ManagedCertificateParams object with the specified parameters.
     * @param string $keyType
     * @param SubjectCertificateParams $params
     * @param int $expiration
     */
    public function __construct(string $keyType, SubjectCertificateParams $params, int $expiration)
    {
        $this->keyType = $keyType;
        $this->subjectParams = $params;
        $this->expiration = $expiration;
    }

    public static function fromProto(\Bloock\ManagedCertificateParams $res): ManagedCertificateParams
    {
        return new ManagedCertificateParams(
            $res->getKeyType(),
            SubjectCertificateParams::fromProto($res->getSubject()),
            $res->getExpiration()
        );
    }

    public function toProto(): \Bloock\ManagedCertificateParams
    {
        $p = new \Bloock\ManagedCertificateParams();
        $p->setKeyType(KeyType::toProto($this->keyType));
        $p->setSubject($this->subjectParams->toProto());
        $p->setExpiration($this->expiration);

        return $p;
    }
}

/**
 * Represents the parameters for importing a certificate.
 */
class ImportCertificateParams
{
    public ?string $password;

    /**
     * Constructs a new ImportCertificateParams object with the specified or not password.
     * @param ?string $keyType
     */
    public function __construct(?string $password = null)
    {
        $this->password = $password;
    }
}

File: languages/python/bloock/entity/key/managed_certificate_params.py
from __future__ import annotations

import bloock._bridge.proto.keys_entities_pb2 as proto
from bloock.entity.key.key_type import KeyType
from bloock.entity.key.subject_certificate_params import SubjectCertificateParams


class ManagedCertificateParams:
    """
    Represents parameters for creating a managed certificate.
    """
    def __init__(
            self,
            key_type: KeyType,
            subject: SubjectCertificateParams,
            expiration: int = 0,
    ) -> None:
        """
        Constructs a ManagedCertificateParams object with the specified parameters.
        :type expiration: object
        :type subject: object
        :type key_type: object
        :rtype: object
        """
        self.key_type = key_type
        """
        Is the type of the key.
        """
        self.subject = subject
        """
        Represents the subject details of the certificate.
        """
        self.expiration = expiration
        """
        Is the number of months until the certificate expiration.
        """

    @staticmethod
    def from_proto(key: proto.ManagedCertificateParams) -> ManagedCertificateParams:
        return ManagedCertificateParams(
            key_type=KeyType.from_proto(key.key_type),
            subject=SubjectCertificateParams.fromProto(
                key.cn, key.c, key.ou, key.o),
            expiration=key.expiration
        )

    def to_proto(self) -> proto.ManagedCertificateParams:
        return proto.ManagedCertificateParams(
            key_type=self.key_type.to_proto(),
            subject=self.subject.to_proto(),
            expiration=self.expiration
        )
